File: Redboost_front/src/app/models/Produit.ts
export class Produit {
  id?: number;
  name: string;
  description?: string;
  price: number;
  stock: number;
  ventes: number;
  poids: number;
  categorie: string;
  dateExpiration?: string;
  image?: string;
  isFavorite: boolean = false; // Added isFavorite property

  constructor(
      name: string,
      price: number,
      stock: number,
      ventes: number,
      poids: number,
      categorie: string,
      dateExpiration: string = '',
      description: string = '',
      image: string = '',
      id?: number
  ) {
      this.id = id;
      this.name = name;
      this.description = description;
      this.price = price;
      this.stock = stock;
      this.ventes = ventes;
      this.poids = poids;
      this.categorie = categorie;
      this.dateExpiration = dateExpiration;
      this.image = image;
      this.isFavorite = false;
  }
}

File: DreamTeam/src/main/java/team/project/redboost/entities/Produit.java
package team.project.redboost.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Produit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;
    private double price;
    private int stock;
    private int ventes;
    private double poids;
    private String categorie;
    private LocalDate dateExpiration;
    @Lob
    private String image; // Will store base64 encoded string
}